### ISSUE TITLE
Webpack compile (non-watching) never exits when Dashboard plugin is included

### ISSUE THREAD

**Issue by rosskevin** [FormidableLabs/webpack-dashboard#125]
First, love the plugin for development!

We use the following for a production build:

```
NODE_ENV=production ./node_modules/.bin/webpack --bail --config webpack.config.js
```

We found that if the `DashboardPlugin` is in webpack, the compile process (a non-watching process) never exits.  The moment we exclude it based on environment, it exits as usual.

While excluding based on the env is workable for us, I don't think that a user expects the dashboard plugin to hang other non-watching processes (and I don't want others spending hours to figure this same thing out).

====================================================================

###### Steps to reproduce the problem

1. Add the Dashboard plugin
1. Compile using `./node_modules/.bin/webpack --bail --config webpack.config.js`

====================================================================

###### More Details

- What operating system are you on? OSX/Ubuntu
- What terminal application are you using? zsh
- What version of webpack-dashboard are you using? 0.2.0
- What is the output of running `echo $TERM`? xterm-256color

**Comment by ayozebarrera** [comment]
I'm having the same issue just compiling with `webpack --optimize-minimize --progress`

**Comment by kenwheeler** [comment]
I highly recommend using the bin based approach, as it fixes this issue and I'll be deprecating the old way soon.

**Comment by rosskevin** [comment]
@kenwheeler what is the `bin based approach`?  I'm using the recommended method as documented here: https://github.com/FormidableLabs/webpack-dashboard#packagejson-recommended

**Comment by rosskevin** [comment]
I see it now, a bit further down in that same section:

```
"scripts": {
    "dev": "webpack-dashboard -- node index.js"
}
```

**Comment by rosskevin** [comment]
Ok, so I have verified my configuration, this bug is present with the recommended configuration.  Here is how I run the processes:

```json
    "start": "webpack-dashboard -- ./node_modules/.bin/webpack-dev-server --config ui/webpack.config.js",
    "build": "npm run clean && NODE_ENV=production ./node_modules/.bin/webpack --bail --config ui/webpack.config.js",
```

When you `npm run build` here (I am not intending to use the dashboard as part of a CI process), it will hang the process.

I think this is unintended and most certainly unexpected.  I think you should reopen as a bug.

**Comment by kenwheeler** [comment]
can you paste your current webpack.config.js in a gist?

**Comment by SteveEdson** [comment]
Getting the same issue, here is my webpack config

```
let webpack = require('webpack'),
    path = require('path'),
    DashboardPlugin = require('webpack-dashboard/plugin');

module.exports = {
    entry: './js/myapp.js',
    output: {
        path: path.resolve(__dirname, '_site/js'),
        filename: 'app.min.js',
        publicPath: './_site/'
    },
    plugins: [
        new DashboardPlugin()
    ],
    resolve: {
        alias: {
            'vue$': 'vue/dist/vue.common.js'
        }
    }
};
```

**Comment by jbielick** [comment]
For me, commenting out `new DashboardPlugin()` allowed the build to finish without hanging.

**Comment by SteveEdson** [comment]
@jbielick yes, but that's not really an option for a Dashboard plugin

**Comment by jbielick** [comment]
@SteveEdson
Oh I thought this was the webpack issue board. Was only trying to narrow down the culprit.

**Comment by tschaub** [comment]
I've proposed a fix for this in #215.  In the meantime, here is a workaround:

```js
// webpack.config.js
const DashboardPlugin = require('webpack-dashboard/plugin');

const devConfig = {
  // rest of config as usual...
  plugins: [
    {
      apply: function() {
        const dashboard = new DashboardPlugin({color: 'cyan'});
        dashboard.apply.apply(dashboard, arguments);
      }
    }
  ]
};
```
